contracts/stellar-interchain-token-service/src/migrate.rs

```diff
@@ -0,0 +1,101 @@
+use stellar_axelar_std::interfaces::CustomMigratableInterface;
+use stellar_axelar_std::{contracttype, soroban_sdk, Address, BytesN, Env, String, Vec};
+use stellar_upgrader::UpgraderClient;
+
+use crate::error::ContractError;
+use crate::flow_limit::current_epoch;
+use crate::storage::TokenIdConfigValue;
+use crate::types::TokenManagerType;
+use crate::{storage, InterchainTokenService};
+
+#[contracttype]
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct CustomMigrationData {
+    pub upgrader_client: Address,
+    pub new_version: String,
+    pub token_ids: Vec<BytesN<32>>,
+    pub new_token_manager_wasm_hash: BytesN<32>,
+    pub new_interchain_token_wasm_hash: BytesN<32>,
+}
+
+pub mod legacy_storage {
+    use stellar_axelar_std::{contractstorage, contracttype, soroban_sdk, BytesN};
+
+    #[contractstorage]
+    enum LegacyDataKey {
+        #[temporary]
+        #[value(i128)]
+        FlowOut { flow_key: FlowKey },
```

Comment:
In hindsight, it would have been better to migrate flow limit in a separate PR since it can be merged independently (and likely quicker) and make this one less complex